photosynthesis_metrics/kid.py

```diff
@@ -1,69 +1,84 @@
 import sys
+from typing import Optional, List, Tuple, Union
+from functools import partial
 
 import numpy as np
-
-from tqdm import tqdm
+import torch
 from sklearn.metrics.pairwise import polynomial_kernel
 
-
-def compute_polynomial_mmd_averages(codes_g, codes_r, n_subsets=50, subset_size=1000,
-                                    ret_var=True, output=sys.stdout, **kernel_args):
-    m = min(codes_g.shape[0], codes_r.shape[0])
+from photosynthesis_metrics.utils import BaseFeatureMetric
+
+def compute_polynomial_mmd_averages(
+    x : np.ndarray,
+    y : np.ndarray,
+    n_subsets : int = 50,
+    subset_size : int = 1000,
+    ret_var : bool = True,
+    **kernel_args
+    ) -> Union[List[np.ndarray], Tuple[List[np.ndarray], List[np.ndarray]]]:
```

Comment:
Wrong return type annotation now
```suggestion
    ) -> Union[Tuple[ndarray, ndarray], ndarray]:
```